src/swap/SwapScreen.tsx

```diff
@@ -525,7 +536,7 @@ export function SwapScreen({ route }: Props) {
         decimalSeparator,
       }),
     }))
-    showMaxCeloSwapWarning()
+    hideOrShowMaxFeeCurrencySwapWarning(fromToken)
```

Comment:
just wanted to check if you considered triggering this from only a useEffect after the fromToken / fromAmount has changed, rather than explicitly triggering it within the function handlers? idk if the useEffect works for all the scenarios, i didn't think it through entirely but it'd be less likely to break if we didn't manually coordinate the warning so it'd be nice if it worked 😀

Reply:
Yes I thought about this, but it seems we only want to show it if the MAX button has been pushed, or when hitting confirm and the amount is greater than the balance.
So here's there's not much alternative, than to leave it in the max button handler.

But I could have kept the `useEffect` for when `fromToken` changes, but decided to move it to the token selection handler given all  the logic is there when they change (along with the network switch), so I felt it belonged better there with the rest.

It's true this could break if we later update the code and change `fromToken` elsewhere. But other things will break anyway.

I agree it's not ideal, I think the whole state logic in this screen could be better handled with `useReducer`.